mantidproject/mantid
# Muon Analysis - Model fitting tab

**Description of work.**
This PR adds a Model Fitting tab to the Muon Analysis interface. It will be hidden initially until we can connect the output fits to a plotting pane. **This means there is no plotting of fitted data when doing model fitting at the moment**

An overview of some new considerations in Model Fitting:
- Selecting a `Results Table` will generate a number of matrix workspaces with a single spectra based on all the different parameter combinations.
- A different fit function is allowed for each parameter dataset because the data will not have 'common features'
- When fitting a sample log which is a string, the strings are replaced with indices 0, 1, 2, etc. so that a fit can be performed. 

**To test:**
Do several fits in the Fitting tab.
Generate a Results workspace on the Results tab
Go to Model Fitting tab. Select the relevant Results workspace (which should already be selected)
The X and Y parameters are already loaded in.
Try Fitting as normal. Look for the output Fit workspace in the ADS and try plotting.




Fixes #31446

*This does not require release notes* because **model fitting is currently hidden.**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
